core/src/main/scala/kafka/admin/AdminUtils.scala

```diff
@@ -272,18 +272,25 @@ object AdminUtils extends Logging with AdminUtilities {
                     numPartitions: Int = 1,
                     replicaAssignmentStr: String = "",
                     checkBrokerAvailable: Boolean = true,
-                    rackAwareMode: RackAwareMode = RackAwareMode.Enforced) {
+                    rackAwareMode: RackAwareMode = RackAwareMode.Enforced,
+                    validateOnly: Boolean = false) {
+    if (zkUtils.pathExists(ZkUtils.ReassignPartitionsPath)) {
+      // We prevent addition partitions while a reassignment is in progress, since
+      // during reassignment there is no meaningful notion of replication factor
+      throw new ReassignmentInProgressException("A partition reassignment is in progress.")
```

Comment:
Why does that matter?

Reply:
In theory, a reassignment could start right after we do this check and before we retrieve the data from ZK. Although it's unlikely for the number of replicas to be changed quickly enough, but generally good to avoid race conditions altogether.